bot/cogs/sync/syncers.py

```diff
@@ -1,235 +1,344 @@
+
+        If a message is given, it is edited to display the prompt and reactions. Otherwise, a new
+        message is sent to the dev-core channel and mentions the core developers role. If the
+        channel cannot be retrieved, return None.
+        """
+        log.trace(f"Sending {self.name} sync confirmation prompt.")
+
+        msg_content = (
+            f'Possible cache issue while syncing {self.name}s. '
+            f'More than {self.MAX_DIFF} {self.name}s were changed. '
+            f'React to confirm or abort the sync.'
         )
 
-    for user in users_to_update:
-        await bot.api_client.put(
-            f'bot/users/{user.id}',
-            json={
-                'avatar_hash': user.avatar_hash,
-                'discriminator': user.discriminator,
-                'id': user.id,
-                'in_guild': user.in_guild,
-                'name': user.name,
-                'roles': list(user.roles)
-            }
+        # Send to core developers if it's an automatic sync.
+        if not message:
+            log.trace("Message not provided for confirmation; creating a new one in dev-core.")
+            channel = self.bot.get_channel(constants.Channels.devcore)
+
+            if not channel:
+                log.debug("Failed to get the dev-core channel from cache; attempting to fetch it.")
+                try:
+                    channel = await self.bot.fetch_channel(constants.Channels.devcore)
+                except HTTPException:
+                    log.exception(
+                        f"Failed to fetch channel for sending sync confirmation prompt; "
+                        f"aborting {self.name} sync."
+                    )
+                    return None
+
+            message = await channel.send(f"{self._CORE_DEV_MENTION}{msg_content}")
+        else:
+            await message.edit(content=msg_content)
+
+        # Add the initial reactions.
+        log.trace(f"Adding reactions to {self.name} syncer confirmation prompt.")
+        for emoji in self._REACTION_EMOJIS:
+            await message.add_reaction(emoji)
+
+        return message
+
+    def _reaction_check(
+        self,
+        author: Member,
+        message: Message,
+        reaction: Reaction,
+        user: t.Union[Member, User]
+    ) -> bool:
+        """
+        Return True if the `reaction` is a valid confirmation or abort reaction on `message`.
+
+        If the `author` of the prompt is a bot, then a reaction by any core developer will be
+        considered valid. Otherwise, the author of the reaction (`user`) will have to be the
+        `author` of the prompt.
+        """
+        # For automatic syncs, check for the core dev role instead of an exact author
+        has_role = any(constants.Roles.core_developer == role.id for role in user.roles)
+        return (
+            reaction.message.id == message.id
+            and not user.bot
+            and (has_role if author.bot else user == author)
```

Comment:
I'm not sure if it's strictly necessary, but this is an elegant solution to limit the check to the author for a manual sync. I guess it prevents someone from accidentally confirming a sync action started by someone else. The point of this comment is purely to say it's an elegant solution.